### DISCUSSION TITLE
RustDesk server running on VM inside network connection issue

### DISCUSSION THREAD

**Discussion by awillerding** [rustdesk/rustdesk#6217] [Q&A]
I've configured RustDesk server to run in a VM on my server and set the firewall on the server to port forward the appropriate ports to my VM server running RustDesk server. The hbbs and hbbr services are running correctly and netstat is showing the ports are connected.  No firewall is running on the VM with the RustDesk server.

On a remote RustDesk client outside my network I've configured the network settings for the ID and Relay servers to point to my server.  The client shows a ready state. 

On a RustDesk client inside my network, I used the internal IP address  of my RustDesk server to configure the ID and Relay servers and it too shows a ready state. 

When I try to connect between these two clients I'm getting "Failed to connect via relay server: Failed to connect to relay server: Please try later" message. 

Is this type of configuration supported and if so then what am I missing to make this work? 

Thanks!

Andrew

**Comment by pdifeo** [comment]
For this reason, it makes sense to have correct aliases for the LAN and the Internet. There is, however, a problem when using multiple relays and the DNS name as per my report

https://github.com/rustdesk/rustdesk-server-pro/issues/102

**Comment by ldellus** [comment]
Hi Andrew,

I have the same problem; did you find a solution?

Thanks,

Laurent

**Comment by ldellus** [comment]
I figured it out. It was a DNS conflict between my pfsense resolver and active directory DNS running on the lan. I turned off resolver, created a slip DNS in active directory DNS (easy, just add a new zone), and made sure active directory DNS has the right forwarders set.

**Reply by pdifeo** [reply]
pfsense dns resolver is based on unbound (https://nlnetlabs.nl/projects/unbound/about/) where you can create a forwarder for a specified zone. You can put the configuration into "Custom options" field of unbound on pfsense

**Comment by ldellus** [comment]
Thanks for the reply. Since I'm using Active Directory DNS, I simply turned off resolver.

One more comment about the issue. The core of my problem was that my RustDesk clients on the LAN were using my .lan domain, while my WAN clients were using my .com domain. I didn't know that both must be configured with the same domain. Once I figured that out, the rest was easy.
